### ISSUE TITLE
AssociativeMemoryLearner.update clips out-of-range/NaN labels into the vocabulary and trains the substituted class with update_applied=True (all-invalid stream reports accuracy 1.0)

### ISSUE THREAD

**Issue by Prabhat1308** [SlopDotCash/asi#458]
## Summary

`AssociativeMemoryLearner.update` (`alberta_framework/core/associative_memory.py`) does `label = jnp.clip(label.astype(jnp.int32), 0, vocab_size - 1)` before anything else. `vocab_size` is documented as "Number of discrete labels/tokens" and validated `>= 2`, but nothing enforces the label domain at runtime, and the repo already ships `safe_discrete_action` in `core/update_safety.py` for exactly this pattern (it returns `(safe_code, valid)` so a caller can fail closed rather than substitute a class). `update` bypasses it and feeds the substituted label into `prior`, `new_row`, `loss`, and `accuracy`, so the fail-closed transaction machinery (`floating_tree_is_finite` / `select_transaction`) never fires — a NaN label becomes class 0 with no trace, `1.9` trains class 1, `-1` trains class 0, `9999` trains class 3.

A run whose entire label stream is out of range (an off-by-one label encoding, a stale vocab, a NaN from an upstream cumulant) publishes `accuracy=1.0`, `loss=0.0`, `updates_applied` all-`True`, `finite=True` through `Step2AssociativeSmokeResult`.

## Falsifier (deterministic, pristine `main` @ `b8a37de`; `vocab_size=4`)

```
label=    -1  update_applied=True  trained class = argmax 0  reported accuracy = 1.0  loss = 0.0000
label=     4  update_applied=True  trained class = argmax 3  reported accuracy = 1.0  loss = 0.0000
label=  9999  update_applied=True  trained class = argmax 3  reported accuracy = 1.0  loss = 0.0000
label=1.9     update_applied=True -> prior written at class 1
label=nan     update_applied=True -> prior written at class 0 prior=[1.0, 0.0, 0.0, 0.0]
label=inf     update_applied=True -> prior written at class 3 prior=[0.0, 0.0, 0.0, 1.0]

run with EVERY label out of range (vocab_size=4, label=4):
  updates_applied all True? True     final-10 accuracy = 1.0  loss = 0.0
```

Expected: an out-of-vocabulary, fractional, or non-finite label is a rejected, neutralized transaction (`update_applied=False`, state unchanged, metrics zeroed) — via `safe_discrete_action(label, vocab_size)` folded into `update_applied`.

## Dedup

Same class, other surface: #398/#399 (PrototypeAgent silently reshapes and reports valid), #447/#448 (IPMNIST runners clamp labels). Open #446 / PR #449 harden the config scalars, not the runtime label; open #456/#457 (mine) fixes the eviction row reset in the same function — the fix PR stacks on it. No issue covers the runtime label domain.

Fix PR follows (failing-test-first, stacked on #457).
